### PR TITLE
Option to add extra utr args in metafiles

### PR DESCRIPTION
**Pull request by kivistikliis**
---
### Purpose of this PR
This PR 
- gathers together the UTR flags for all testplatforms, generalizing the default flags needed for editmode/playmode/standalone, with the option to override their values. Platform specific flags can be appended in their representative cmd files similar to before
- allows to specify additional UTR flags per project per testplatform in metafiles, and to create 'new' testplatforms, by choosing a base (editmode/playmode/standalone), optionally renaming it (e.g. how is done for playmode_XR), and specifying the additional args. This is handy for setting up other testplatforms with ease, or let projects use additional set of flags (such as Hybrids do)
- timeout of test platforms is now configured in project metafiles
- removed osx_metal_split as it is and will be unused

Updated docs https://github.com/Unity-Technologies/Graphics/tree/yamato/extra-args/.yamato/docs#project_namemetafile-project-jobs-configuration

---
### Testing status

**Yamato**: (Select your branch):
Nightly https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics/tree/yamato%252Fextra-args/.yamato%252F_abv.yml%2523all_project_ci_nightly_trunk

**Comments to reviewers:** 
There are lots of changes, so to pinpoint the essentials:
- the utr flags are reordered in all ymls, so we should carefully check that all the flags are in fact what they were before
- the flags are now gathered in ruamel/jobs/shared/utr_utils.py

### CODE REVIEW

File: .yamato/ruamel/jobs/projects/commands/osx_metal.py [hunk 1]
```diff
@@ -1,45 +1,89 @@
...
+def cmd_editmode(project_folder, platform, api, test_platform_args):
+
+    utr_args = utr_editmode_flags(
+        testproject=f'/Users/bokken/{REPOSITORY_NAME}/{TEST_PROJECTS_DIR}/{project_folder}',
+        editor_location=f'/Users/bokken/.Editor',
+        artifacts_path=f'/Users/bokken/{REPOSITORY_NAME}/{TEST_PROJECTS_DIR}/{project_folder}/{PATH_TEST_RESULTS}'
+    )
+    utr_args.extend(test_platform_args)
+    return  _cmd_base(project_folder, platform["components"], utr_args)
+    # base = _cmd_base(project_folder, platform["components"])
```
Comment (theo-pnv) [inline .yamato/ruamel/jobs/projects/commands/osx_metal.py:32]: Is it intended to keep the old code commented?

File: .yamato/ruamel/metafile_parser.py [hunk 2]
```diff
@@ -78,13 +78,27 @@ def _unfold_platforms(metafile, shared):
 def _unfold_test_platforms(metafile, shared, root_keys=[]):
     '''Retrieves test platform details from shared metafile, corresponding to the specific metafile. 
     Returns the new 'test_platforms' section.'''
+    # tps = []
+    # for tp in metafile.get("test_platforms", []):
```
Comment (theo-pnv) [inline .yamato/ruamel/metafile_parser.py:82]: Is it intended to keep this code commented?

File: .yamato/universal-android-opengles3.yml [hunk 3]
```diff
@@ -49,7 +49,7 @@ Build_Universal_Android_OpenGLES3_Player_trunk:
       -  mklink /d WindowsEditor\Data\PlaybackEngines\AndroidPlayer\OpenJDK %JAVA_HOME% || exit 0
       -  mklink /d WindowsEditor\Data\PlaybackEngines\AndroidPlayer\SDK %ANDROID_SDK_ROOT% || exit 0
       -  mklink /d WindowsEditor\Data\PlaybackEngines\AndroidPlayer\NDK %ANDROID_NDK_ROOT% || exit 0
-      -  utr --suite=playmode --platform=Android --testproject=TestProjects\UniversalGraphicsTest --editor-location=WindowsEditor --artifacts_path=test-results --player-save-path=players --scripting-backend=il2cpp --timeout=1800 --build-only
+      -  utr --suite=playmode --platform=Android --testproject=TestProjects\UniversalGraphicsTest --editor-location=WindowsEditor --artifacts_path=test-results --timeout=1200 --player-save-path=players --build-only --scripting-backend=il2cpp
```
Comment (theo-pnv) [inline .yamato/universal-android-opengles3.yml:52]: It seems timeout=1200 has slipped through for this build step (was at 1800 before)

Comment (theo-pnv) [reply .yamato/universal-android-opengles3.yml:52]: (Same for fast-trunk and CUSTOM-REVISION)

Comment (theo-pnv) [reply .yamato/universal-android-opengles3.yml:52]: (And for the other APIs 😄)

File: .yamato/universal_hybrid-android-vulkan.yml [hunk 4]
```diff
@@ -49,7 +49,7 @@ Build_Universal_Hybrid_Android_Vulkan_Player_trunk:
       -  mklink /d WindowsEditor\Data\PlaybackEngines\AndroidPlayer\OpenJDK %JAVA_HOME% || exit 0
       -  mklink /d WindowsEditor\Data\PlaybackEngines\AndroidPlayer\SDK %ANDROID_SDK_ROOT% || exit 0
       -  mklink /d WindowsEditor\Data\PlaybackEngines\AndroidPlayer\NDK %ANDROID_NDK_ROOT% || exit 0
-      -  utr --suite=playmode --platform=Android --testproject=TestProjects\UniversalHybridTest --extra-editor-arg="-executemethod" --extra-editor-arg="SetupProject.ApplySettings" --extra-editor-arg="vulkan" --editor-location=WindowsEditor --artifacts_path=test-results --player-save-path=players --scripting-backend=il2cpp --timeout=1800 --build-only --compilation-errors-as-warnings
+      -  utr --suite=playmode --platform=Android --testproject=TestProjects\UniversalHybridTest --editor-location=WindowsEditor --artifacts_path=test-results --timeout=1200 --player-save-path=players --build-only --compilation-errors-as-warnings --scripting-backend=il2cpp --extra-editor-arg="-executemethod" --extra-editor-arg="SetupProject.ApplySettings" --extra-editor-arg="vulkan"
```
Comment (theo-pnv) [inline .yamato/universal_hybrid-android-vulkan.yml:52]: Same for hybrid tests (timeout=1200)

**Review by theo-pnv — CHANGES_REQUESTED** [review]
Great work!

File: .yamato/ruamel/jobs/projects/commands/osx_metal.py (hunk 1 above)
Comment (kivistikliis) [reply .yamato/ruamel/jobs/projects/commands/osx_metal.py:32]: Ah yes, it is not, thanks for noticing!

File: .yamato/universal-android-opengles3.yml (hunk 3 above)
Comment (kivistikliis) [reply .yamato/universal-android-opengles3.yml:52]: Good catch! Double-checking all of these now

**Review by sophiaaar — APPROVED** [review]

**Review by theo-pnv — APPROVED** [review]

**Comment by sophiaaar** [comment]
no changes to yml other than rearranging args

**Pull request merged**